tensorboard/plugins/image/metadata.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Internal information about the images plugin."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import json
+
+import tensorflow as tf
+
+
+PLUGIN_NAME = 'images'
+
+
+ImageMetadata = collections.namedtuple('ImageMetadata', ())
+
+
+def create_summary_metadata(display_name, description):
```

Comment:
As I went to implement this change, my opinion shifted slightly. You're right that the _abstraction_ is trivial, but the _convenience_ is nice, as is the consistency. This code is used in four or five places: `image.summary.op`, `image.summary.pb`, `data_compat._migrate_image`, and `data_compat_test.test_image` (and it really should be used in `image_summary_test`, too; I'll fix that.) It seems wise to provide a single entry point for all of these so that the metadata is consistent—wouldn't you agree?

We can definitely still use a binary proto instead of a JSON object.